### PR TITLE
Add Jupyter action for CodeMirror indentAuto command

### PR DESCRIPTION
**Pull request by gnestor**
Closes https://github.com/jupyter/notebook/issues/2591

Adds a shortcut for CodeMirror’s `indentAuto` command that doesn’t collide with special characters on French/German keyboards (https://github.com/jupyter/notebook/pull/2535) and also adds it to both the cell editor and text/file editor.

### CODE REVIEW

**Comment by takluyver** [comment]
How confident are we that this shortcut is not already used for anything in some browser? Generally, whenever we add a Ctrl- shortcut, we get complaints that it overrides some other shortcut that people like.

**Comment by gnestor** [comment]
Similar sentiment expressed by @Carreau at https://github.com/jupyter/notebook/issues/2591#issuecomment-354978107:

> The typical reasonig for shortcut is the following:
>
> No shortcut should use Alt (or Shift but in a lesser measure) by default. None. Ever.
>
> Many non-english keyboard need Alt and or Shift to type characters. So Shift and Alt are not available.
>
> Typically, french keyboard use Shift for Numbers. So Cmd-Shift-[ is actually "Cmd-5" on a french keyboard... which is taken, and on French keyboard typing [ actually require pressing Alt-Shift-5... so I'm not even sure how that will conflict resolve, or if this could trigger ghost shortcuts.
>
> Hence I would prefer not to do it. What we can (and maybe should) do, is provide a set of "nice but not enabled by default" keymap that are a click away, and provide a "Querty-layout" extra, "Azerty layout extra", "Cyrillic extra", where we can go as crazy as we like.

**Comment by gnestor** [comment]
Would it be possible to add some of these "nice to have but not necessary" CodeMirror commands to the notebook actions, allowing people to access `indentAuto` from the command palette as well as set a custom keyboard shortcut for it? In this instance, it could be as simple as:

```js
// Get selected cell
var selected_cell = notebook.get_selected_cell();
// Execute a CM command
selected_cell.code_mirror.execCommand('indentAuto');
```

**Comment by takluyver** [comment]
I think that makes sense, yep. :-)

File: notebook/static/edit/js/editor.js [hunk 1]
```diff
@@ -82,10 +82,10 @@ function(
             "Tab": "indentMore",
             "Shift-Tab" : "indentLess",
             "Cmd-/" : "toggleComment",
-            "Ctrl-/" : "toggleComment",
+            "Ctrl-/" : "toggleComment"
         },
         indentUnit: 4,
-        theme: "ipython",
+        theme: "default",
```
Comment (takluyver) [inline notebook/static/edit/js/editor.js]: Not sure why this change is part of this PR?

**Comment by takluyver** [comment]
@gnestor if you want to get rid of the editor theme change, I think it's fine to merge this PR for 5.3.

File: notebook/static/edit/js/editor.js (hunk 1 above)
Comment (gnestor) [reply notebook/static/edit/js/editor.js]: Initially, the intent was to make the CodeMirror config (shortcuts, theme, etc.) consistent between code cells in the notebook and the file editor. The code cells use the "default" theme and the editor uses the "ipython" theme. Is this intentional?

Comment (takluyver) [reply notebook/static/edit/js/editor.js]: I don't know. I don't have any particular objection if you think it's the right thing to do, but I think it would have been better to do it as a separate PR so it could be discussed separately.

Git blame shows @minrk set it to use the IPython theme three years ago in 0efd335553afe8d84d4a16aed9fc37db643390f2 . Min, do you have any recollection whether there was a need for that?

Comment (minrk) [reply notebook/static/edit/js/editor.js]: Oof, that's asking a lot of my foggy memory. Maybe it had to do with how magics were highlighted? Try highlighting some IPython-specific syntax and see if it works the same. If it looks fine, I've no objection.

**Comment by gnestor** [comment]
Ok, I'm moving the syntax stuff to another PR. This is ready to merge 👍

**Comment by takluyver** [comment]
Thanks :-)
